src/components/Chat/ChatSession.js

```diff
@@ -263,15 +267,43 @@ class ChatSession {
     return this.client.sendDeliveredReceipt(messageId, options);
   }
 
-  addOutgoingMessage(data) {
-    const message = modelUtils.createOutgoingTranscriptItem(
+  alterOutgoingMessageForViewsIfRequired(data) {
+    /**
+     * if using guides, expect the response to be of type INTERACTIVE_RESPONSE
+     * else take the no match found branch of the view
+     */
+    const lastIncomingMessageIdx = this._findLastMessageInTranscript(Direction.Incoming, this.transcript);
+    if (lastIncomingMessageIdx >= 0) {
+      const lastIncomingMessage = this.transcript[lastIncomingMessageIdx];
+      try {
+        const lastIncomingMessageData = JSON.parse(lastIncomingMessage.content.data);
+
+        if (lastIncomingMessageData.templateType === InteractiveMessageType.VIEW_RESOURCE &&
+          data.type !== ContentType.MESSAGE_CONTENT_TYPE.INTERACTIVE_RESPONSE) {
+          let temp_message = {
+            action: " ", // empty string is not allowed
```

Comment:
This hits the no match branch. That's expected behavior right now, yes?

Reply:
yes, expectation is to hit no match when view is not interacted with

Reply:
Basically, even if user has not interacted with the View eg: form. We will end up sending `interactive.response` with text message what user entered in Chat ?
Is this message used anywhere in flows ?

Reply:
QQ. Is it requirement to always send `interactive.response` message after receiving interactive message ?
What happens if I just send a normal message and not `interactive.response` type message ?

Reply:
Yes, an interactive.response is required to exit a show view block. Sending a normal message like plaintext will have no effect. The message the user entered in the chat will be available under $.Views.ViewResultData, the customer making the flow can optionally use this if required. The flow does not use the message typed by the customer and takes the noMatch route of the block solely based on the empty action.